components/CategoriesBar/CategoryLink.tsx

```diff
@@ -1,27 +1,48 @@
 import type { Category } from '@prezly/sdk';
 import { getCategoryUrl, getLocalizedCategoryData } from '@prezly/theme-kit-core';
 import { useCurrentLocale, useGetLinkLocaleSlug } from '@prezly/theme-kit-nextjs';
+import { useWindowSize } from '@react-hookz/web';
 import classNames from 'classnames';
 import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+import type { RefObject } from 'react';
 
 import { ButtonLink } from '@/ui';
 
 import styles from './CategoriesBar.module.scss';
 
 interface Props {
     category: Category;
+    containerRef: RefObject<HTMLDivElement>;
+    onIsOverflowing: (isOverflowing: boolean) => void;
 }
 
-function CategoryLink({ category }: Props) {
+const MORE_BUTTON_WIDTH = +styles.MORE_BUTTON_WIDTH.replace('px', '');
+
+function CategoryLink({ category, containerRef, onIsOverflowing }: Props) {
+    const { asPath } = useRouter();
+    const { width: windowWidth } = useWindowSize();
     const currentLocale = useCurrentLocale();
     const { name } = getLocalizedCategoryData(category, currentLocale);
     const getLinkLocaleSlug = useGetLinkLocaleSlug();
-    const { asPath } = useRouter();
-
+    const ref = useRef<HTMLAnchorElement>(null);
     const isActive = asPath.includes(getCategoryUrl(category, currentLocale));
 
+    useEffect(() => {
+        const container = containerRef.current;
+        const element = ref.current;
+        if (container && element) {
+            const { right: containerRight } = container.getBoundingClientRect();
```

Comment:
> Previously we were using character count to identify if categories are overflowing or not, I have used [getBoundingClientRect](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) instead as it is more accurate.

> It solves the problem though, but I'm not sure why we didn't go with this approach in the first place. Maybe I'm missing something? It's a lot of hacks to get it to work though.

I don't think there was a particular reason to go with character count. One issue with your new approach is that the categories bar has a brief content flash right after the page load. While character count was not super accurate, it could be calculated on the server side, preventing this flash.

I can also see another issue in that while the categories bar is updated correctly when you shrink the screen, hiding the categories, it doesn't reveal them back when you increase the screen width.